packages/itwinui-react/src/core/Menu/MenuItem.tsx

```diff
@@ -187,7 +190,7 @@ export const MenuItem = React.forwardRef((props, forwardedRef) => {
       case 'Enter':
       case ' ':
       case 'Spacebar': {
-        !disabled && onClick?.(value);
+        onActivate();
```

Comment:
Can you remind me what was the problem with `useClick`?

_(Continued from [previous conversation](https://github.com/iTwin/iTwinUI/pull/1942#discussion_r1572609858))_

Reply:
After adding the click trigger, I noticed that it required two clicks to close a submenu. But turns out that it was expected behavior ([floating-ui/floating-ui#1893 (comment)](https://redirect.github.com/floating-ui/floating-ui/issues/1893#issuecomment-1250161220)).

I worked around this by manually toggling the visible state in `onClick()` ([code](https://github.com/iTwin/iTwinUI/pull/1942/commits/b3c54a164a3f9e4c2760a4623031d6ae63f9e4de#diff-4410321a0ef9f99d8d2c38b7907baf2e10f7fdf8efcff3a9ca5245db96d35db9R223)).